aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/build/BuildTriggerStepTest.java

```diff
@@ -165,4 +170,25 @@ public void cancelBuildQueue() throws Exception {
         j.assertLogContains("branch=release extra=true", ds.getBuildByNumber(3));
     }
 
+    @Issue("JENKINS-26123")
+    @Test public void noWait() throws Exception {
+        j.createFreeStyleProject("ds").setAssignedLabel(Label.get("nonexistent"));
+        WorkflowJob us = j.jenkins.createProject(WorkflowJob.class, "us");
+        us.setDefinition(new CpsFlowDefinition("build job: 'ds', wait: false"));
+        j.assertBuildStatusSuccess(us.scheduleBuild2(0));
```

Comment:
Scheduling the `WorkflowJob` will succeed. Its attempt to schedule the `FreeStyleProject` will fail. This causes `scheduleBuild2` to return `null`, which needs to make the step fail, which is the bug I fixed.